tests/functional/models/test_application.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+import json
+from datetime import datetime
+
+from tests.helper import TestHelper
+
+
+class TestApplication(unittest.TestCase):
+
+    helper = None
+    resin = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.helper = TestHelper()
+        cls.resin = cls.helper.resin
+
+    def tearDown(self):
+        # Wipe all apps after every test case.
+        self.helper.wipe_application()
+
+    def test_create(self):
+        # should be rejected if the device type is invalid
+        with self.assertRaises(self.helper.resin_exceptions.InvalidDeviceType):
+            self.resin.models.application.create('FooBar', 'Foo')
+
+        # should be rejected if the name has less than three characters
+        with self.assertRaises(Exception) as cm:
+            self.resin.models.application.create('Fo', 'Raspberry Pi 2')
+        self.assertIn('It is necessary that each app name that is of a user (Auth), has a Length (Type) that is greater than or equal to 4', cm.exception.message)
+
+        # should be able to create an application
+        app = json.loads(self.resin.models.application.create('FooBar', 'Raspberry Pi 2').decode('utf-8'))
+        self.assertEqual(app['app_name'], 'FooBar')
+
+    def test_get_all(self):
+        # given no applications, it should return empty list.
+        self.assertEqual(self.resin.models.application.get_all(), [])
+
+        # given there is an application, it should return a list with length 1.
+        self.resin.models.application.create('FooBar', 'Raspberry Pi 2')
+        self.assertEqual(len(self.resin.models.application.get_all()), 1)
```

Comment:
As there's only one application, you could also check the `app_name`, as you do in the next test?

Would it also be worth creating a test for adding two applications, to check that `get_all()` can cope with more than one result? ;) And you could check e.g.
```python
all_applications = self.resin.models.application.get_all()
self.assertEqual(len(all_applications), 2)
self.assertNotEqual(all_applications[0]['app_name'], all_applications[1]['app_name'])
```

Reply:
added this to the test case.